middleware/grpc/interceptor.go

```diff
@@ -110,7 +117,13 @@ This call would result in an authorization resource with the following structure
 If the value of "address" is itself a message, all of its fields are included.
 */
 func (m *Middleware) WithResourceFromFields(fields ...string) *Middleware {
+	if len(fields) == 1 && fields[0] == "*" {
```

Comment:
Perhaps you can move this check to `messageResourceMapper` so it can handle either an explicit list of fields or `"*"` to copy the entire message. Then we won't need a separate `reqMessageResourceMapper`.